app/domain/authentication/authn_jwt/fetch_issuer_value.rb

```diff
@@ -0,0 +1,131 @@
+require 'uri'
+
+module Authentication
+  module AuthnJwt
+
+    # FetchIssuerValue command class is responsible to fetch the issuer secret value,
+    # in order to validate it later against the JWT token claim
+    FetchIssuerValue ||= CommandClass.new(
+      dependencies: {
+        resource_class: ::Resource,
+        fetch_secrets: ::Conjur::FetchRequiredSecrets.new,
+        logger: Rails.logger
+      },
+      inputs: %i[authenticator_input]
+    ) do
+      extend(Forwardable)
+      def_delegators(:@authenticator_input, :service_id, :authenticator_name,
+                     :account)
+
+      def call
+        @logger.debug(LogMessages::Authentication::AuthnJwt::FetchingIssuerConfigurationValue.new)
+        fetch_issuer_value
+        @logger.debug(LogMessages::Authentication::AuthnJwt::FetchedIssuerValueFromConfiguration.new)
+
+        @issuer_value
+      end
+
+      private
+
+      def fetch_issuer_value
```

Comment:
please add a comment explaining the logic. specifically the order in which we look at each place